roles/keycloak/vars/main.yml

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) 2022 VEXXHOST, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+_keycloak_helm_values:
+  # Note(okozachenko1203): Mysql vendor is not supported by bitnami helm chart. As a workaround,
+  #                        we have to define jdbc connection string explicitly along side
+  #                        `externalDatabase` helm values.
+  extraEnvVars:
+    - name: KC_PROXY
+      value: edge
+    - name: KC_DB
+      value: mysql
+    - name: KC_DB_URL
+      value: "jdbc:mysql://{{ openstack_helm_endpoints.oslo_db.hosts.default }}.openstack:3306/{{ keycloak_database_name }}"
+    - name: KC_DB_USERNAME
+      value: "{{ keycloak_database_username }}"
+    - name: KC_DB_PASSWORD
+      valueFrom:
+        secretKeyRef:
+          key: db-password
+          name: keycloak-externaldb
+  command:
+    - /opt/keycloak/bin/kc.sh
+    - --verbose
+    - start
+    - --auto-build
+    - --http-enabled=true
+    - --http-port=8080
+    - --hostname-strict=false
+    - --spi-events-listener-jboss-logging-success-level=info
+    - --spi-events-listener-jboss-logging-error-level=warn
+    - --transaction-xa-enabled=false
+  auth:
+    adminPassword: "{{ keycloak_admin_password }}"
+    adminUser: "{{ keycloak_admin_username | default('admin') }}"
+  externalDatabase:
+    host: "{{ openstack_helm_endpoints.oslo_db.hosts.default }}.openstack"
+    port: 3306
+    database: "{{ keycloak_database_name }}"
+    user: "{{ keycloak_database_username }}"
+    password: "{{ keycloak_database_password }}"
+  image:
+    registry: "{{ atmosphere_images['keycloak'] | vexxhost.kubernetes.docker_image('domain') }}"
+    repository: "{{ atmosphere_images['keycloak'] | vexxhost.kubernetes.docker_image('path') }}"
+    tag: "{{ atmosphere_images['keycloak'] | vexxhost.kubernetes.docker_image('tag') }}"
+  ingress:
+    enabled: true
+    ingressClassName: "{{ keycloak_ingress_class_name }}"
+    annotations:
+      cert-manager.io/cluster-issuer: atmosphere
+    hostname: "{{ keycloak_host }}"
+    tls: true
```

Comment:
my concern with using their built in ingress thing, is that users cannot override this with common overrides

we are using `openstack_helm_ingress` and users are kinda using the overrides there, .. maybe we can create role `ingress` and then `openstack_helm_ingress` uses it and then we can make this role us it as well?

Reply:
ok.

But i think managing ingresses out of the helm chart has a downside.
In osh helm charts, resource names are static, not generated from the release name, you know. But in other normal helm charts, resource names are generated by their own template and includes helm release name.
It means we have to check how names are generated in the helm chart manifest and follow the same template to get the service name required for ingress spec.
Also when we upgrade the helm chart version, if there is any changes in the template or ingress version or sth related to ingress or service or annotation, we have to mirror them by using our own ingress role.

Btw, I created a new role for ingress and used it. Not a swiss-army knife for now. I think we can improve to support more complex specs like extraHosts, extraPaths, etc. in the future when we need it.